apps/sms/test/marionette/lib/storage.js

```diff
@@ -15,40 +15,10 @@
          * generated during test (e.g. send or receive new message).
          */
         setMessagesStorage: function(threads, uniqueMessageIdCounter) {
-          client.executeScript(function(threads, uniqueMessageIdCounter) {
-            var STORAGE_NAME = 'messagesDB';
-
-            var recipientToThreadId = new Map();
-
-            var messageMap = new Map();
-
-            var threadMap = new Map(threads.map(function(thread) {
-              recipientToThreadId.set(thread.participants[0], thread.id);
-
-              // Messages should be placed into dedicated map to simplify
-              // message-based manipulations.
-              if (thread.messages) {
-                thread.messages = new Set(
-                  thread.messages.map(function(message) {
-                    messageMap.set(message.id, message);
-
-                    return message.id;
-                  })
-                );
-              }
-
-              return [thread.id, thread];
-            }));
-
-            window.wrappedJSObject.TestStorages.setStorage(STORAGE_NAME, {
-              threads: threadMap,
-              messages: messageMap,
-              recipientToThreadId: recipientToThreadId,
-              uniqueMessageIdCounter: uniqueMessageIdCounter
-            });
-
-            window.wrappedJSObject.TestStorages.setStorageReady(STORAGE_NAME);
-          }, [threads || [], uniqueMessageIdCounter || 0]);
+          this.setStorage('messagesDB', {
```

Comment:
I moved storage preparation code to mock itself as it's the only mock that can use this data directly. and to clean this code a bit.